### PR TITLE
[NEW] Livechat WebRTC call settings

### PR DESCRIPTION
**Pull request by djcruz93**
## Proposed changes (including videos or screenshots)

Add new functionality to enable webRTC call in livechat. It also fixes Jitsi settings in Livechat. Earlier if Jitsi is enabled from Video Conference settings, it got enabled automatically for livechat bypassing the livechat enable video call toggle button. Now the flow is same for Jitsi and WebRTC to enable calling in Livechat ----> Enable WebRTC/Jitsi then select call provider from admin/Omnichannel.

Before:
image
This toggle button is removed and Jitsi provider can now be selected from call provider select field inside Livechat Call accordion.

Now:
https://user-images.githubusercontent.com/51796498/124357117-750c8980-dc37-11eb-8ee9-c55c66e79c00.mp4

### CODE REVIEW

File: app/theme/client/imports/general/base.css [hunk 1]
```diff
@@ -239,7 +239,7 @@ button {
 	.rc-old .main-content-flex {
 		flex: 1 0 auto !important; /* 1 */
 
-		height: auto !important; /* 1 */ 
+		height: auto !important; /* 1 */
```
Comment (murtaza98) [inline app/theme/client/imports/general/base.css:242]: Didn't this this change, doesn't seem relevant to this PR. If its by mistake, please remove it

File: app/videobridge/client/tabBar.tsx [hunk 2]
```diff
@@ -49,11 +49,12 @@ addAction('video', ({ room }) => {
 
 	const enabledChannel = useSetting('Jitsi_Enable_Channels');
 	const enabledTeams = useSetting('Jitsi_Enable_Teams');
+	const enabledLiveChat = useSetting('Livechat_call_provider') === t('Jitsi');
```
Comment (murtaza98) [inline app/videobridge/client/tabBar.tsx:52]:
```suggestion
	const enabledLiveChat = useSetting('Livechat_call_provider') === 'Jitsi';
```

File: app/webrtc/server/settings.js [hunk 3]
```diff
@@ -1,24 +1,34 @@
 import { settings } from '../../settings';
 
 settings.addGroup('WebRTC', function() {
+	this.add('WebRTC_Enabled', false, {
```
Comment (murtaza98) [inline app/webrtc/server/settings.js:4]: Couple of points here
1. I see you've defined this setting here however I don't see this setting being used anywhere except within the new file `app/webrtc/client/tabBar.tsx` which you've created in this PR. Since you're introducing this new setting here, we'll need to make sure we enforce this setting in all the relevant places in the codebase.
2. We'll be needing a DB migration for this one. I didn't see any migrations for this in your PR.

File: app/livechat/server/config.js [hunk 4]
```diff
@@ -329,16 +329,6 @@ Meteor.startup(function() {
 		i18nLabel: 'Continuous_sound_notifications_for_new_livechat_room',
 	});
 
-	settings.add('Livechat_videocall_enabled', false, {
```
Comment (murtaza98) [inline app/livechat/server/config.js:332]: You've removed this setting here but I can still find references to this within the codebase. Could you please check this and modify those references accordingly?

Secondly, we'll also need another migration for this setting. Users who already have this and jitsi setting on, should have the `Jitsi` option selected within our new setting i.e. `Livechat_call_provider`

File: app/livechat/server/config.js [hunk 5]
```diff
@@ -329,16 +329,6 @@ Meteor.startup(function() {
...
-		type: 'boolean',
-		group: 'Omnichannel',
-		section: 'Livechat',
-		public: true,
-		i18nLabel: 'Videocall_enabled',
```
Comment (murtaza98) [inline app/livechat/server/config.js:337]: We can remove these translations if they are not being used anywhere else

File: app/livechat/server/config.js [hunk 6]
```diff
@@ -555,4 +545,18 @@ Meteor.startup(function() {
 		i18nLabel: 'How_long_to_wait_to_consider_visitor_abandonment',
 		i18nDescription: 'Time_in_seconds',
 	});
+
+	settings.add('Livechat_call_provider', 'none', {
```
Comment (murtaza98) [inline app/livechat/server/config.js:549]: Lets call this `Omnichannel_call_provider` instead, since this setting would apply for other channels as well

File: app/livechat/server/config.js [hunk 7]
```diff
@@ -555,4 +545,18 @@ Meteor.startup(function() {
...
+		type: 'select',
+		public: true,
+		group: 'Omnichannel',
+		section: 'Livechat Call',
```
Comment (murtaza98) [inline app/livechat/server/config.js:553]: Same goes here as well, by adding Livechat we're restricting the scope of this setting. Lets call it `Calls` or `Audio and Video calls`

Comment (murtaza98) [reply app/livechat/server/config.js:553]: Also I didn't find any translations for this section within the i18n folder. Please add those as well

File: app/webrtc/client/tabBar.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+import { useSetting } from '../../../client/contexts/SettingsContext';
+import { addAction } from '../../../client/views/room/lib/Toolbox';
+import { useTranslation } from '../../../client/contexts/TranslationContext';
+
+addAction('webRTCVideo', () => {
+	const t = useTranslation();
+	const enabled = useSetting('WebRTC_Enabled') && (useSetting('Livechat_call_provider') === t('WebRTC'));
+
+	const handleClick = () => { window.open('/', '_blank'); };
```
Comment (murtaza98) [inline app/webrtc/client/tabBar.tsx:11]: image
New warning getting introduced bcs of this code. Please resolve them

File: app/webrtc/client/tabBar.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+	return useMemo(() => (enabled ? {
+		groups: ['live'],
+		id: 'webRTCVideo',
+		title: 'WebRTC_Call',
+		icon: 'phone',
+		action: handleClick,
+		full: true,
+		order: 4,
+	} : null), [enabled, t]);
```
Comment (murtaza98) [inline app/webrtc/client/tabBar.tsx:21]: image
New warning getting introduced bcs of this code. Please resolve them

**Review by murtaza98 — CHANGES_REQUESTED** [review]
Hello @djcruz93 I've added my initial review here. Please take a look

File: app/theme/client/imports/general/base.css [hunk 10]
```diff
@@ -269,4 +269,4 @@ button {
 
 /*
  1. The `!important` directive is necessary to override components' autogenerated styles.
- */
\ No newline at end of file
+ */
```
Comment (murtaza98) [inline app/theme/client/imports/general/base.css:272]: Same goes for here. 
> Didn't get this change, doesn't seem relevant to this PR. If its by mistake, please remove it

File: app/livechat/server/methods/getInitialData.js [hunk 11]
```diff
@@ -75,7 +75,7 @@ Meteor.methods({
 		info.offlineUnavailableMessage = initSettings.Livechat_offline_form_unavailable;
 		info.displayOfflineForm = initSettings.Livechat_display_offline_form;
 		info.language = initSettings.Language;
-		info.videoCall = initSettings.Livechat_videocall_enabled === true && initSettings.Jitsi_Enabled === true;
+		info.videoCall = initSettings.Omnichannel_call_provider === 'Jistsi' && initSettings.Jitsi_Enabled === true;
```
Comment (murtaza98) [inline app/livechat/server/methods/getInitialData.js:78]:
```suggestion
		info.videoCall = initSettings.Omnichannel_call_provider === 'Jitsi' && initSettings.Jitsi_Enabled === true;
```

**Review by murtaza98 — COMMENTED** [review]
LGTM! There's one small type to be fixed though. @djcruz93 let me know when that's fixed so I can approve this PR

**Comment by djcruz93** [comment]
@murtaza98 I have pushed the typo changes. You may approve this PR.

**Review by murtaza98 — APPROVED** [review]

**Comment by murtaza98** [comment]
@djcruz93 I've merged this PR. Please rebase the other PR's accordingly. Thanks
